### ISSUE TITLE
OpenFGA upgrade from 1.3.0 to 1.5.4 has DB migration issue.

### ISSUE THREAD

**Issue by Charish-Spotdraft** [openfga/openfga#2111]
### Checklist

- [x] I have looked into the [README](https://github.com/openfga/openfga/blob/main/README.md) and have not found a suitable solution or answer.
- [x] I have looked into the [documentation](https://openfga.dev/docs) and have not found a suitable solution or answer.
- [x] I have searched the [issues](https://github.com/openfga/openfga/issues) and have not found a suitable solution or answer.
- [x] I have upgraded to the [latest version](https://github.com/openfga/openfga/releases/latest) of OpenFGA and the issue still persists.
- [x] I have searched the [OpenFGA Community](https://openfga.dev/community) and have not found a suitable solution or answer.
- [x] I agree to the terms within the [OpenFGA Code of Conduct](https://github.com/openfga/.github/blob/main/CODE_OF_CONDUCT.md).

## Summary
After upgrading OpenFGA from version 1.3.0 to 1.5.4, requests to the /check endpoint fail with the error message: `"type <my_entity> not found"`. This issue occurs despite the entity type being defined and working correctly in version 1.3.0.

## Steps to Reproduce

- used the [docker compose file](https://github.com/openfga/openfga/blob/main/docker-compose.yaml)(updated it to have volumes for postgres mounted). updated file: [docker-compose.yml.txt](https://github.com/user-attachments/files/17835607/docker-compose.yml.txt) (remove `.txt` to use it, had to add it as github would not let me upload `.yml` files).
- initially use v1.3.0 of openfga i.e `image: openfga/openfga:v1.3.0` in the docker compose file
- `docker-compose up -d`
- add sample data: store.fga.yaml can look like [contract.fga.yaml.txt](https://github.com/user-attachments/files/17840004/contract.fga.yaml.txt)
- make `/read` and `/check` calls for the tuple and observe that there are no issues
- `docker-compose down`
- update the docker compose file to set the version to 1.5.4 i.e `image: openfga/openfga:v1.5.4`
- `docker-compose up -d`
- `/read` call is successful but `/check` throws an error with message `"type 'contract' not found"`

There seems to be an issue with migrate where the DB is not gracefully migrated so the entities are not recognized in newer version.

### Relevant Links
Had raised in slack community channel ([thread](https://cloud-native.slack.com/archives/C06G1NNH47N/p1730099521667279))  earlier
### OpenFGA version

1.5.4

### How are you running OpenFGA?

In Kubernetes

### What datastore are you using?

Postgres

### OpenFGA Flags

NA

### Logs

logline for the failed `/check` call
```
{"grpc_service": "openfga.v1.OpenFGAService", "grpc_method": "Check", "grpc_type": "unary", "user_agent": "PostmanRuntime/7.42.0", "raw_request": {"store_id":"01JD6G2SCP9KVR3T14EVFM76XK","tuple_key":{"user":"user:1","relation":"viewer","object":"contract:1"},"contextual_tuples":null,"authorization_model_id":"","trace":false,"context":null}, "raw_response": {"code":"validation_error","message":"type 'contract' not found"}, "peer.address": "127.0.0.1:59482", "request_id": "aea60c53-ea73-4301-823f-529884983990", "store_id": "01JD6G2SCP9KVR3T14EVFM76XK", "authorization_model_id": "01JD6G34V18EXDVMS77WV1X42A", "grpc_code": 2000, "error": "rpc error: code = Code(2000) desc = type 'contract' not found"}
```

**Comment by Charish-Spotdraft** [comment]
cc: @aaguiarz

**Comment by miparnisari** [comment]
Hi @Charish-Spotdraft! I'm curious, is there a specific reason why you are attempting to migrate to 1.5.4 and not the latest version?

In any case, this is a known issue that has been fixed in [v1.6.2](https://github.com/openfga/openfga/releases/tag/v1.6.2).

Let me know if this helps :)

**Comment by Charish-Spotdraft** [comment]
Hello @miparnisari ,

> is there a specific reason why you are attempting to migrate to 1.5.4 and not the latest version?

No specific reason was trying to use the version with `enable-list-users` and had only checked the change log till 1.5.4 so was not aware of the fix in 1.6.2.
thanks for the help, this solves my issue.
closing the bug.
